src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,45 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<T, K> {
+    private T firstObject;
+    private K secondObject;
 
+    private Pair(T firstObject, K secondObject) {
+        this.firstObject = firstObject;
+        this.secondObject = secondObject;
+    }
+
+    public T getFirst() {
+        return firstObject;
+    }
+
+    public K getSecond() {
+        return secondObject;
+    }
+
+    public static <T, K> Pair<T, K> of(T first, K second) {
+        return new Pair<>(first, second);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Pair<?, ?> pair = (Pair<?, ?>) o;
+        return (firstObject == null && pair.firstObject == null
```

Comment:
```suggestion
        return (firstObject == pair.firstObject
```
you can also simplify this

Reply:
done